group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -1580,6 +1601,9 @@ public void run() {
                     if (!future.isDone()) {
                         operationTimeout = new OperationTimeout(tp, this, writeTimeout.toMillis());
                         timer.add(operationTimeout);
+                        
+                        // Only update when the records were appended to the local log.
+                        writeTimestamp = time.milliseconds();
```

Comment:
In this patch, we always write, I think.

Reply:
I'm not sure I follow